spec/crypto/cert_spec.rb

```diff
@@ -590,6 +590,17 @@
         expect(subject.issuer).to eq(ca_cert.subject)
       end
     end
+
+    context "when the ca: keyword argument is given" do
+      subject do
+        Ronin::Support::Crypto::Cert.generate(key: rsa_key, ca: true)
+      end
+
+      it "must add 'basicConstraints' => ['CA:TRUE', true] to the extensions" do
+        expect(subject.extension_names).to eq(['basicConstraints'])
+        expect(subject.extension_value('basicConstraints')).to eq("CA:TRUE")
+      end
```

Comment:
Probably needs an additional spec where both `ca: true` and `extensions: {...}` are given to ensure that `ca: true` does not override `extensions:`. You can adapt the "when the extensions: keyword is given" specs to test `extensions` contains both `basicConstraints` and other `extensions:` values.